cabot/cabotapp/tests/test_api.py

```diff
@@ -362,16 +364,21 @@ def test_counter_reset(self):
         self.assertEqual(json.loads(response.content), expected_body)
 
     def test_check_should_run_when_activity_counter_disabled(self):
-        self.http_check.use_activity_counter = False
-        self.http_check.activity_counter.count = 0
+        self._set_activity_counter(False, 0)
         self.assertTrue(self.http_check.should_run())
 
     def test_check_should_run_when_activity_counter_positive(self):
-        self.http_check.use_activity_counter = True
-        self.http_check.activity_counter.count = 1
+        self._set_activity_counter(True, 1)
         self.assertTrue(self.http_check.should_run())
 
     def test_check_should_not_run_when_activity_counter_zero(self):
+        self._set_activity_counter(True, 0)
+        self.assertFalse(self.http_check.should_run())
+
+    def test_check_should_not_run_when_activity_counter_missing(self):
```

Comment:
Add a test for the situation this is correcting? (`should_run` called when activity counter doesn't exist?)